framework/src/util/Utils.cpp

```diff
@@ -392,7 +392,8 @@ int GetLastErrorNo()
 std::string GetLastErrorStr()
 {
 #ifdef US_PLATFORM_POSIX
-  return std::string(strerror(errno));
+  char* errorString = strerror(errno);
```

Comment:
I just realized that we are using the thread-unsafe `strerror` function here, but `GetLastErrorStr()` may be called concurrently. I will open a separate issue for this.